transport.go

```diff
@@ -80,46 +81,65 @@ func addWebhookRequestDecoder(config transportConfig) kithttp.DecodeRequestFunc
 
 	// if no validators are given, we accept anything.
 	if config.v == nil {
-		config.v = AlwaysValid()
+		config.v = append(config.v, webhook.AlwaysValid())
 	}
 
 	return func(c context.Context, r *http.Request) (request interface{}, err error) {
 		requestPayload, err := io.ReadAll(r.Body)
 		if err != nil {
 			return nil, err
 		}
-		var wr WebhookRegistration
+		var v1 *webhook.RegistrationV1
+		var v2 *webhook.RegistrationV2
+		var whreq addWebhookRequest
 
-		err = json.Unmarshal(requestPayload, &wr)
-		if err != nil {
-			var e *json.UnmarshalTypeError
-			if errors.As(err, &e) {
-				return nil, &erraux.Error{Err: fmt.Errorf("%w: %v must be of type %v", errFailedWebhookUnmarshal, e.Field, e.Type), Code: http.StatusBadRequest}
-			}
-			return nil, &erraux.Error{Err: fmt.Errorf("%w: %v", errFailedWebhookUnmarshal, err), Code: http.StatusBadRequest}
-		}
-
-		webhook := wr.ToWebhook()
-		err = config.v.Validate(webhook)
-		if err != nil {
-			return nil, &erraux.Error{Err: err, Message: "failed webhook validation", Code: http.StatusBadRequest}
-		}
-
-		wv.setWebhookDefaults(&webhook, r.RemoteAddr)
+		opts := config.v
 
 		var partners []string
 		partners, err = extractPartnerIDs(config, c, r)
 		if err != nil && !config.disablePartnerIDs {
 			return nil, &erraux.Error{Err: err, Message: "failed getting partnerIDs", Code: http.StatusBadRequest}
 		}
 
-		return &addWebhookRequest{
-			owner: getOwner(r.Context()),
-			internalWebook: InternalWebhook{
-				Webhook:    webhook,
+		err = json.Unmarshal(requestPayload, &v1)
+		if err == nil {
+			err = opts.Validate(&v1)
+			if err != nil {
+				return nil, &erraux.Error{Err: err, Message: "failed webhook validation", Code: http.StatusBadRequest}
+			}
+			wv.setWebhookDefaults(v1, r.RemoteAddr)
+			reg := RegistryV1{
 				PartnerIDs: partners,
-			},
-		}, nil
+				Webhook:    *v1,
+			}
+
+			whreq.internalWebook = reg
+		} else {
+			err = json.Unmarshal(requestPayload, &v2)
+			if err == nil {
+				err = config.v.Validate(&v2)
+				if err != nil {
+					return nil, &erraux.Error{Err: err, Message: "failed webhook validation", Code: http.StatusBadRequest}
+				}
+				reg := RegistryV2{
+					PartnerIds:   partners,
+					Registration: *v2,
+				}
+				whreq.internalWebook = reg
+			}
+		}
+
+		if err != nil {
+			var e *json.UnmarshalTypeError
+			if errors.As(err, &e) {
+				return nil, &erraux.Error{Err: fmt.Errorf("%w: %v must be of type webhook.RegistrationV1 or webhook.RegistrationV2", errFailedWebhookUnmarshal, e.Field), Code: http.StatusBadRequest}
+			}
+			return nil, &erraux.Error{Err: fmt.Errorf("%w: %v", errFailedWebhookUnmarshal, err), Code: http.StatusBadRequest}
+		}
+		whreq.owner = getOwner(r.Context())
+
+		return &whreq, nil
+
```

Comment:
I would recommend to move the v2 unmarshaling outside of the v1 unmarshal if/else block.

I would also suggest to change the order of the error checking for both v1/v2 unmarshal to make this section easier on the eyes, i.e. move things around from line 104-142.